### ISSUE TITLE
Cloudwatch alarms

### ISSUE THREAD

**Issue by mirkods** [horike37/serverless-step-functions#163]
# Feature Proposal

## Description
Hi there,
congratulations for this very good serverless pluging.

I'd like to have the possibility to create a Cloudwatch alarm based on Step Functions executions state in order to subscribe a lambda function to a specific topic.
<a href="https://github.com/ACloudGuru/serverless-plugin-aws-alerts">This plugin</a> works well for lambda functions so I think that something similar applied to Step Functions could be really interesting.

**Comment by theburningmonk** [comment]
@mirkods what are you thinking? I've been thinking of something similar, and this is what I had in mind:

```
stepFunctions:
  stateMachines:
    processPayment:
      name: something-something
      definition:
        ...
      alarms:
        topics:
          ok: ${self:service}-${opt:stage}-alerts-ok
          alarm: ${self:service}-${opt:stage}-alerts-alarm
          insufficientData: ${self:service}-${opt:stage}-alerts-insufficientData
        metrics:
          - executionsTimeOut
          - executionsFailed
          - executionsAborted
          - executionThrottled
```

where the plugin would auto-generate CW twitchy (as in, trip the alarm when there's any error) alarms for the 4 metrics that correspond to failure cases in CloudWatch

![image](https://user-images.githubusercontent.com/546969/51995208-2f699680-24aa-11e9-9394-6258aaffbaff.png)

**Comment by mirkods** [comment]
@theburningmonk this is exactly what I meant and I think that this part can be an important improvement for this module.

**Comment by theburningmonk** [comment]
Awesome, I've been thinking about similar lines as well, as we constantly have to create those alarms either manually or use the ACG plugin, which doesn't offer an opinionated set of defaults that works for step functions.

**Comment by theburningmonk** [comment]
this has been implemented by #167
